src/information_theory/hamming.lean

```diff
@@ -121,49 +124,51 @@ section has_zero
 variables [Π i, has_zero (β i)] [Π i, has_zero (γ i)]
 
 /-- The Hamming weight function to the naturals. -/
-def hamming_norm (x : Π i, β i) : ℕ := hamming_dist x 0
-
-/-- Corresponds to `dist_zero_right`. -/
-lemma hamming_dist_zero_right (x : Π i, β i) : hamming_dist x 0 = hamming_norm x := rfl
-
-/-- Corresponds to `dist_zero_left`. -/
-lemma hamming_dist_zero_left : hamming_dist (0 : Π i, β i) = hamming_norm :=
-funext $ λ x, by rw [hamming_dist_comm, hamming_dist_zero_right]
+def hamming_norm (x : Π i, β i) : ℕ := (univ.filter (λ i, x i ≠ 0)).card
 
 /-- Corresponds to `norm_nonneg`. -/
-lemma hamming_norm_nonneg {x : Π i, β i} : 0 ≤ hamming_norm x := hamming_dist_nonneg
+@[simp] lemma hamming_norm_nonneg {x : Π i, β i} : 0 ≤ hamming_norm x := zero_le _
 
 /-- Corresponds to `norm_zero`. -/
-@[simp] lemma hamming_norm_zero : hamming_norm (0 : Π i, β i) = 0 := hamming_dist_self _
```

Comment:
Doesn't this old proof still work?

Reply:
I was trying to avoid any kind of abusing defeq and separate the norm and dist proofs, I think?